File: flutter_sdk/lib/src/client/chat_client_config.dart
import '../session/session_store.dart';
import '../utils/logger.dart';

/// Configuration for [ChatClient].
///
/// Example:
/// ```dart
/// final config = ChatClientConfig(
///   baseUrl: 'https://api.example.com/api',
///   accessToken: oidcToken,
///   userId: 'user-uuid',
/// );
/// final client = await ChatClient.initialize(config);
/// ```
class ChatClientConfig {
  /// Base URL of the `rust-e2e-chat-api` server (no trailing slash).
  /// e.g. `"https://api.example.com/api"`
  final String baseUrl;

  /// OIDC JWT access token for the current user.
  final String accessToken;

  /// The OIDC `sub` claim value identifying the user.
  final String userId;

  /// Previously registered device ID. Pass null to trigger auto-registration
  /// (generates a new key bundle and registers the device).
  final String? deviceId;

  /// Custom session store. Defaults to [FlutterSecureSessionStore] on mobile/
  /// desktop and [InMemorySessionStore] on web/server.
  final SessionStore? sessionStore;

  /// If true (default), the SDK connects to the WebSocket immediately after
  /// [ChatClient.initialize] returns.
  final bool autoConnect;

  /// How many days before the Signed Pre-Key is automatically rotated.
  /// Default: 7 days.
  final int signedPrekeyRotationDays;

  /// Controls the verbosity of internal SDK logging.
  /// Default: [LogLevel.warning].
  final LogLevel logLevel;

  const ChatClientConfig({
    required this.baseUrl,
    required this.accessToken,
    required this.userId,
    this.deviceId,
    this.sessionStore,
    this.autoConnect = true,
    this.signedPrekeyRotationDays = 7,
    this.logLevel = LogLevel.warning,
  });
}

File: web_sdk/src/client/chat-client-config.ts
// src/client/chat-client-config.ts

import type { SessionStore } from '../session/session-store.js';
import type { LogLevel } from '../utils/logger.js';

/**
 * Configuration for ChatClient.create().
 */
export interface ChatClientConfig {
  /** Base URL of the API server, e.g. "https://api.example.com" */
  baseUrl: string;
  /** OIDC JWT access token */
  accessToken: string;
  /** OIDC sub claim — the caller's user ID */
  userId: string;
  /** Device ID — if undefined, a new device will be registered */
  deviceId?: string;
  /** Custom session store. Defaults to IndexedDbSessionStore in browsers, MemorySessionStore elsewhere */
  sessionStore?: SessionStore;
  /** Connect automatically on create(). Default: true */
  autoConnect?: boolean;
  /** Days between SignedPreKey rotations. Default: 7 */
  signedPrekeyRotationDays?: number;
  /** Log level. Default: 'warn' */
  logLevel?: LogLevel;
  /** Custom WebSocket constructor (Node.js < 21 without global WebSocket) */
  WebSocketImpl?: typeof WebSocket;
}
